### PR TITLE
Add com.system76.Cosmic.BaseApp

### PR DESCRIPTION
**Pull request by jackpot51**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
A COSMIC BaseApp that Includes the just command runner, pop-icon-theme, and cosmic-icons which most COSMIC applications need.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by jackpot51** [comment]
This replaces the COSMIC runtime proposed in #5654, and will simplify COSMIC application submissions #5648 and #5649. Requesting review by @bbhtt, who recommended this approach.

File: com.system76.CosmicBaseApp.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+    "id": "com.system76.CosmicBaseApp",
+    "branch": "1.0.0-alpha.2",
```
Comment (bbhtt) [inline com.system76.CosmicBaseApp.json:3]: Does these components have an API/ABI involved ? Or apps are supposed to always use the latest version of theme from here ?

If so I'd drop the `branch`.

**Comment by jackpot51** [comment]
Can I use finish_args here to allow COSMIC applications by default to access the settings daemon dbus service and the cosmic config folder?

File: com.system76.CosmicBaseApp.json (hunk 1 above)
Comment (jackpot51) [reply com.system76.CosmicBaseApp.json:3]: There are no ABI/API versioned components here as of now.

Comment (jackpot51) [reply com.system76.CosmicBaseApp.json]: I have removed `branch`

**Comment by wash2** [comment]
`separate-locales: false` is included in the elementary base app. Should we disable that as well?

**Comment by bbhtt** [comment]
> `separate-locales: false` is included in the elementary base app. Should we disable that as well?

Currently no locale is produced here, but sure it's fine to add it.

**Comment by bbhtt** [comment]
bot, build com.system76.CosmicBaseApp

**Comment by hfiguiere** [comment]
there should be a script to cleanup stuff like `just` that is not needed at runtime.

also what toolkit does COSMIC uses?

**Comment by jackpot51** [comment]
COSMIC uses libcosmic, which is statically compiled. Just is left in as it is the default build system for COSMIC applications.

**Comment by jackpot51** [comment]
I think I can add the just cleanup because it will be available for the build of an app based on this, right?

**Comment by hfiguiere** [comment]
> I think I can add the just cleanup 

if you do that, there is virtually no point in having that package.

**Comment by bbhtt** [comment]
> there should be a script to cleanup stuff like just that is not needed at runtime.

Yea I'd had a script like this https://github.com/flathub/com.riverbankcomputing.PyQt.BaseApp/blob/d86580d472384d877a366d5aba3b61e79bcce1f3/com.riverbankcomputing.PyQt.BaseApp.yaml#L20-L28

Apps just add it in `cleanup-commands` https://github.com/flathub/com.riverbankcomputing.PyQt.BaseApp?tab=readme-ov-file#how-to-use like this:

```
cleanup-commands:
  - /app/cleanup-BaseApp.sh
```

> I think I can add the just cleanup because it will be available for the build of an app based on this, right?

No, cleanups run at the end of the build, so if you add it here the baseapp won't have `just`

**Comment by jackpot51** [comment]
I have added a cleanup script

**Comment by bbhtt** [comment]
bot, build com.system76.CosmicBaseApp

**Comment by jackpot51** [comment]
The errors are  "finish-args-not-defined" and "toplevel-no-command". I don't believe either finish-args or a command should be defined, they are not for io.elementary.BaseApp.

**Comment by jackpot51** [comment]
Should I submit a PR for exceptions?

**Comment by jackpot51** [comment]
It looks like there is a problem with the ID being used. It needs to end with ".BaseApp". https://github.com/flathub-infra/flatpak-builder-lint/blob/6ca3f8131fed842123f82e1bc4b8f03bbef9069d/flatpak_builder_lint/checks/appid.py#L29

Is it okay if I change the ID to be `com.system76.Cosmic.BaseApp`?

**Comment by jackpot51** [comment]
I have changed the ID and verified locally that it passes the linter.

**Comment by jackpot51** [comment]
I have added flathub.json in order to skip appstream checks: https://github.com/flathub/io.elementary.BaseApp/blob/branch/juno-23.08/flathub.json

**Comment by bbhtt** [comment]
Ah yea you need to end the ID with `.BaseApp` that's how the linter skips checks on it that don't matter. I forgot about that.

The `flathub.json` is unused, those parameters were dropped.

bot, build com.system76.Cosmic.BaseApp

**Comment by jackpot51** [comment]
I have deleted flathub.json

**Comment by jackpot51** [comment]
Yes, it is

**Comment by jackpot51** [comment]
@barthalion if it helps, I can add a `branch`. It was recommended to drop it as we don't have any ABI components, in https://github.com/flathub/flathub/pull/5655#discussion_r1783226509

**Comment by jackpot51** [comment]
@bbhtt or @barthalion, what else do I need to do for this?

**Comment by bbhtt** [comment]
PR is ok, I think barthalion is just busy, usually barthalion handles the merges.

**Pull request closed without merging**